dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs

```diff
@@ -0,0 +1,459 @@
+///      The model then reasons about the chat history and then calls functions again or generates the final response.
+///      This approach is fully automated and requires no manual intervention from the caller. The automatic invocation mode is enabled by default.
+///    * Manual invocation mode returns all function calls requested by the AI model to the AI API caller. The caller is fully responsible
+///      for the invocation phase where they may decide which function to call, how to handle exceptions, call them in parallel or sequentially, etc.
+///      The caller then adds the function results/exceptions to the chat history and returns it to the model, which reasons about it
+///      and then calls functions again or generates the final response. This invocation mode provides more control over the function invocation phase to the caller.
+///      To enable manual invocation, the caller needs to set the `autoInvoke` parameter to `false` when specifying either <see cref="FunctionChoiceBehavior.Auto"/>
+///      or <see cref="FunctionChoiceBehavior.Required"/> in the <see cref="PromptExecutionSettings"/>.
+///
+///    SK supports only sequential invocation of functions in the automatic invocation mode at the moment. The concurrent invocation mode is only possible
+///    in the manual invocation mode.
+/// </summary>
+public class FunctionCalling(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// This example demonstrates usage of <see cref="FunctionChoiceBehavior.Auto"/> that advertises all kernel functions to the AI model and invokes them automatically.
+    /// </summary>
+    [Fact]
+    public async Task RunPromptWithAutoFunctionChoiceBehaviorAdvertisingAllKernelFunctionsInvokedAutomaticallyAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto() };
+
+        Console.WriteLine(await kernel.InvokePromptAsync("Given the current time of day and weather, what is the likely color of the sky in Boston?", new(settings)));
+
+        // Expected output: "Boston is currently experiencing a rainy day, hence, the likely color of the sky in Boston is grey."
+    }
+
+    /// <summary>
+    /// This example demonstrates usage of <see cref="FunctionChoiceBehavior.Required"/> that advertises only one function to the AI model and invokes it automatically.
+    /// </summary>
+    [Fact]
+    public async Task RunPromptWithRequiredFunctionChoiceBehaviorAdvertisingOneFunctionInvokedAutomaticallyAsync()
+    {
+        Kernel kernel = CreateKernel();
+
+        KernelFunction getWeatherFunction = kernel.Plugins.GetFunction("HelperFunctions", "GetWeatherForCity");
+
+        OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Required(functions: [getWeatherFunction]) };
```

Comment:
Even though this example uses function array, it's important to add an example with multiple `required` functions, so we can share it with users who were asking how to pass multiple `required` functions in settings.

Reply:
It's not fully supported yet hence only one function for now.